src/main/java/org/takes/facets/auth/social/PsTwitter.java

```diff
@@ -0,0 +1,183 @@
+        throws IOException {
+        return Collections.singleton(
+            PsTwitter.fetch(this.token())
+        ).iterator();
+    }
+
+    @Override
+    public Response exit(final Response response,
+        final Identity identity) {
+        return response;
+    }
+
+    /**
+     * Get user name from Github, with the token provided.
+     * @param token Github access token
+     * @return The user found in Github
+     * @throws IOException If fails
+     */
+    private static Identity fetch(final String token) throws IOException {
+        final String uri = new Href(
+                "https://api.twitter.com/1.1/account/verify_credentials.json"
+        )
+            .with("access_token", token)
+            .toString();
+        return PsTwitter.parse(
+            new JdkRequest(uri)
+                .header("accept", "application/json")
+                .fetch().as(RestResponse.class)
+                .assertStatus(HttpURLConnection.HTTP_OK)
+                .as(JsonResponse.class).json().readObject()
+        );
+    }
+
+    /**
+     * Retrieve Github access token.
+     * @return The token
+     * @throws IOException If failed
+     */
+    private String token()
+        throws IOException {
+        final String uri = new Href("https://api.twitter.com/oauth2/token")
+            .with("grant_type", "client_credentials")
+            .toString();
+        return new JdkRequest(uri)
+            .method("POST")
+                .header(
+                        "Content-Type",
+                        "application/x-www-form-urlencoded;charset=UTF-8"
+            )
+                .header(
+                        "Authorization",
+                        "Basic ".concat(
+                                PsTwitter.prepareAuthKey(
+                                this.app, this.key
+                    )
+                )
+            )
+            .fetch().as(RestResponse.class)
+            .assertStatus(HttpURLConnection.HTTP_OK)
+            .as(JsonResponse.class)
+            .json().readObject().getString("access_token");
+    }
+
+    /**
+     * Make identity from JSON object.
+     * @param json JSON received from Github
+     * @return Identity found
+     */
+    private static Identity parse(final JsonObject json) {
+        final ConcurrentMap<String, String> props =
+            new ConcurrentHashMap<String, String>(json.size());
+        props.put("name", json.getString("name"));
+        props.put("picture", json.getString("profile_image_url"));
+        return new Identity.Simple(
+            String.format("urn:twitter:%d", json.getInt("id")), props
+        );
+    }
+
+    /**
+     * Encode consumerKey and consumerSecret and prepare authorization key.
+     * @param consumerkey Twitter consumerKey
+     * @param consumersecret Twitter consumerSecret
+     * @return String authorization key
+     */
+    private static String prepareAuthKey(final String consumerkey,
```

Comment:
don't think we would need this method at all as soon as migrated on DatatypeConverter